cairo-libs/packages/cometbft/src/types.cairo

```diff
@@ -176,16 +184,19 @@ impl BlockIdFlagAsProtoMessage of ProtoMessage<BlockIdFlag> {
         }
     }
 
-    fn decode_raw(ref self: BlockIdFlag, ref context: DecodeContext) {
-        let mut var = Default::<u32>::default();
-        var.decode_raw(ref context);
-        match var {
-            0 => self = BlockIdFlag::Unknown,
-            1 => self = BlockIdFlag::Absent,
-            2 => self = BlockIdFlag::Commit,
-            3 => self = BlockIdFlag::Nil,
-            _ => panic!("invalid block Id flag"),
+    fn decode_raw(ref context: DecodeContext) -> Option<BlockIdFlag> {
+        let var: Option<u32> = context.decode_raw();
+        if var.is_none() {
+            return Option::None;
```

Comment:
Here again.

Reply:
There are few other places, this happens. Search for `return Option::None` on your branch. I believe there ~10 cases including cases like

```cairo
        if let Result::Ok(v_u64) = varint {
            let v_u32 = v_u64.try_into();
            if let Option::Some(_) = v_u32 {
                return v_u32;
            } else {
                return Option::None;
            }
        } else {
            return Option::None;
        }
```

and

```cairo
        if let Option::Some(n) = num {
            return Option::Some(decode_2_complement_64(@n));
        } else {
            return Option::None;
        }
```

Reply:
[4df6ada](https://github.com/informalsystems/ibc-starknet/pull/281/commits/4df6adac7233bcf46a6c866af2513151acfed0a7)